library/ecp_curves.c

```diff
@@ -5181,6 +5183,102 @@ int mbedtls_ecp_mod_p256_raw(mbedtls_mpi_uint *X, size_t X_limbs)
 
 #endif /* MBEDTLS_ECP_DP_SECP256R1_ENABLED */
 
+#if defined(MBEDTLS_ECP_DP_SECP384R1_ENABLED)
+/*
+ * Fast quasi-reduction modulo p384 (FIPS 186-3 D.2.4)
+ */
+static int ecp_mod_p384(mbedtls_mpi *N)
+{
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    size_t expected_width = 2 * ((384 + biL - 1) / biL);
```

Comment:
1. This is `required_width` not `expected_width`
2. The doc for `mbedtls_ecp_mod_p384_raw()` says required size is `768 / biL`, but I don't see `768` in this calculation (note: 384 is 3 * 128, so even with `biL` == 128 (which it is never) the `+ biL - 1) / biL` thing wouldn't be needed. But if it is needed, please keep 768 in there, i.e. `(768 + biL - 1) / biL`. But `768 / biL` with a comment that 768 = 3 * 256, so no need to worry about partial limbs even if limb size was 256 bits would be fine

Reply:
1. Again that is ported as is from the prototype, and has been used in all the similar curves. I would rather we keep it consistent as is.
2. This wording a [request ](https://github.com/Mbed-TLS/mbedtls/pull/7222#discussion_r1129597577)coming from the previous round of review. But I agree that is is confusing.

I will reword it to something like

~~~~~~
Must have exact limb size that stores a 768-bit MPI
 *                          (double the bitlength of the modulus).
~~~~~~

The size check is not performed in the X parameter but the `X_limbs` which is enforced to be less than `2* 386bits` . 

In the public method, a grow operation ensures, that the static method will be called with appropriate sizes:

`MBEDTLS_MPI_CHK(mbedtls_mpi_grow(N, expected_width));`